go-controller/pkg/ovn/policy.go

```diff
@@ -1179,6 +1209,23 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) error {
 			return fmt.Errorf("failed to create default port groups and acls for policy: %s/%s, error: %v",
 				policy.Namespace, policy.Name, err)
 		}
+		defer func() {
+			if err != nil && len(nsInfo.networkPolicies) == 0 {
```

Comment:
the check needs to be after you acquire the new nsInfo lock on line 1214. So the check should really just be on line 1220 if len(nsInfo.networkPolicies)==0